sqlglot/errors.py

```diff
@@ -19,3 +19,7 @@ class ParseError(ValueError):
 
 class TokenError(ValueError):
     pass
+
+
+class OptimizeError(ValueError):
```

Comment:
clearly not affected by this PR but we should add a top level error for better handling outside of sqlglot